docs/contributing/contributing-rules.md

```diff
@@ -0,0 +1,136 @@
+---
+slug: contributing-to-semgrep-rules-repository
+description: "This article outlines how to contribute to Semgrep rules repository.."
+---
+
+# Contributing to Semgrep rules
+
+## Introduction
+
+There are two ways in which you can contribute to the Semgrep rules repository:
+
+<dl>
+    <dt>For users of Semgrep App</dt>
+    <dd>Contribute to the Semgrep rules repository through Semgrep App. This workflow is recommended. See <a href="#contributing-through-semgrep-app-recommended"> Contributing through Semgrep App (recommended)</a>. This workflow creates the necessary pull request for you and streamlines the whole process.
+</dd>
+    <dt>For contributors to the repository through GitHub</dt>
+    <dd>Contribute to the Semgrep rules repository through a pull request. See the <a href="#contributing-through-github"> Contributing through GitHub</a> section for detailed information.</dd>
+</dl>
+
+### Contributing through Semgrep App (recommended)
+
+To contribute to the Semgrep rules repository through Semgrep App, follow these steps:
+1. Go to [Semgrep App Editor](https://semgrep.dev/orgs/-/editor).
+2. Click **Create New Rule**.
+3. Make one of the following steps:
+    - Create a new rule and test code, and then click **Save**. Note: The test file must contain at least one true positive and one true negative test case in order to get approved
+    - Select a rule from a category in **Semgrep Registry**. Modify the rule or test code, click **Save**, and then **Fork**.
+4. Click **Share**.
+
+This workflow automatically creates a pull request in the GitHub [rules repository](https://github.com/returntocorp/semgrep-rules). Find more about the rules repository by reading the [Rule writing](#rule-writing) and [Tests](#tests) sections.
+
+### Contributing through GitHub
+
+Fork our repository and make a pull request; we'll contact you about signing our Contributor License Agreement (CLA). Install pre-commit (see [installing pre-commit](#installing-pre-commit)) and make a pull request to the [rules repository](https://github.com/returntocorp/semgrep-rules) with two files:
+1. The semgrep pattern (.yml)
```

Comment:
nit: We prefer `.yaml`, but it doesn't really matter

Reply:
@minusworld it should be YAML actually. (Search for YAML [here](https://developers.google.com/style/filenames?hl=en)). It slipped through... Thanks! (I appreciate thorough reviews.)